post_report.py

```diff
@@ -0,0 +1,86 @@
+
+from pywell.entry_points import run_from_cli
+from pywell.notify_slack import notify_slack
+import requests
+
+from get_pending_accounts import get_pending_accounts
+from filter_emails_by_membership import filter_emails_by_membership
+
+
+DESCRIPTION = 'Post a report on pending accounts from Mobilize.'
+
+ARG_DEFINITIONS = {
+    'DB_HOST': 'Database host IP or hostname',
+    'DB_PORT': 'Database port number',
+    'DB_USER': 'Database user',
+    'DB_PASS': 'Database password',
+    'DB_NAME': 'Database name',
+    'MOBILIZE_API_KEY': 'API key for mobilize.io',
+    'MOBILIZE_API_SECRET': 'API secret for mobilize.io',
+    'MOBILIZE_API_ROOT': 'Root URL for mobilize.io API',
+    'MOBILIZE_DEFAULT_GROUP_ID': 'The first group members request to join',
+    'SLACK_WEBHOOK': 'Web hook URL for Slack.',
+    'SLACK_CHANNEL': 'Slack channel to send to.',
+    'VERBOSE': 'Give verbose output if set'
+}
+
+REQUIRED_ARGS = [
+    'DB_HOST', 'DB_PORT', 'DB_USER', 'DB_PASS', 'DB_NAME',
+    'MOBILIZE_API_KEY', 'MOBILIZE_API_SECRET', 'MOBILIZE_API_ROOT',
+    'MOBILIZE_DEFAULT_GROUP_ID', 'SLACK_WEBHOOK', 'SLACK_CHANNEL'
+]
+
+
+def post_report(args) -> list:
+    accounts = get_pending_accounts(args)
+    args.EMAILS = ','.join([account.get('email', '') for account in accounts])
+    approved_emails = filter_emails_by_membership(args)
+    approved_accounts = [
+        account for account in accounts
+        if account.get('email', '') in approved_emails
+    ]
+    declined_accounts = [
+        account for account in accounts
+        if account.get('email', '') not in approved_emails
+    ]
+    if len(approved_accounts):
+        args.SLACK_MESSAGE_TEXT = "The following pending Mobilize accounts "
+        args.SLACK_MESSAGE_TEXT += "should be *approved* "
+        args.SLACK_MESSAGE_TEXT += "(subscribed to MoveOn's email list):\n"
+        args.SLACK_MESSAGE_TEXT += "\n".join([
+            "• %s (%s)" % (
+                account.get('name', '[NO NAME]'),
+                account.get('email', '[NO EMAIL]')
+            )
+            for account in approved_accounts
```

Comment:
Nit - not a fan of defining variables after they are used (unless this is idiomatic for Python)

Reply:
I think this is idiomatic for Python, at least as long as we're open to using list comprehensions: https://docs.python.org/3.7/tutorial/datastructures.html#list-comprehensions

It's possible to avoid list comprehensions to avoid variables definitions happening after use, e.g.:
```message_parts = []
for account in approved_accounts:
    message_parts.append("• %s (%s)" % (
          account.get('name', '[NO NAME]'),
          account.get('email', '[NO EMAIL]')
    ))
args.SLACK_MESSAGE_TEXT += "\n".join(message_parts)
```

It's also possible to do less logic inside the list comprehension, so the gap between use and definition is minimal, e.g.:

`args.SLACK_MESSAGE_TEXT += "\n".join([account_description(account) for account in accounts])`

I personally like that last option best, so I'm going to switch to that for now. I'm also open to following another style choice but I think that should be something we decide across all our code and everyone writing that code, as 2 people agreeing on something in a single PR isn't much of a standard.